src/xpcc/architecture/platform/driver/core/cortex/driver.xml

```diff
@@ -38,11 +39,22 @@
 		<!-- stm32f7 -->
 		<template device-family="f7" out="linkerscript.ld">stm32/stm32_idtcm.ld.in</template>
 
-		<template>startup.c.in</template>
-		<template>delay.cpp.in</template>
-		<template>syscalls.cpp.in</template>
+		<!-- dealing with C++ -->
 		<static>cxxabi.cpp</static>
+		<!-- dealing with newlib -->
+		<static>syscalls.c</static>
+		<!-- startup script -->
+		<template>startup.c.in</template>
 		<static>default_handler.sx</static>
+		<!-- everything to do with dynamic memory -->
+		<static>heap_table.c</static>
+		<template>heap_newlib.c.in</template>
```

Comment:
> @dergraaf's library-builder uses Python as well, especially for the module options. So does Conan.

The problem with using fully fledged python is that IDE and other tool support is quite difficult. While CMake and make have reasonable IDE support, there is no such thing for SCons, imho because it is very hard to parse all dependencies.